src/persist/src/lib.rs

```diff
@@ -10,6 +10,11 @@
 //! Persistence for Materialize dataflows.
 
 #![warn(missing_docs)]
+#![warn(
+    clippy::cast_possible_truncation,
```

Comment:
@benesch this doesn't actually do anything but I can't add these to bin/check because they're not true for the rest of the codebase. any suggestions on how to keep them enforced for the persist crate?

Reply:
Wait, really? I feel like this should work? It worked for me in `pgrepr` at least: https://github.com/MaterializeInc/materialize/blob/613ffedbe4188fc9afbd080a3ea0bf8ddf552561/src/pgrepr/src/lib.rs#L19

Maybe these aren't actually the right clippy lints? Not sure.

Reply:
It triggers them if I run `cargo clippy -p persist`, but `bin/check` seems to override this so I was unconvinced that adding one would cause CI to fail. If it only shows up when someone manually runs `cargo clippy` outside of bin/check, I'm not convinced that it'll do anything. I didn't actually try pushing it in the bad state, so maybe something else in CI would catch it?

Reply:
`bin/check` is additive, so it should only override what you've got here if it fights with anything explicitly listed in `bin/check`.

Reply:
I just tested this again, in case I was crazy the first time, and running ./bin/check after adding a violation doesn't fail. Specifically, the following diff on top of what's in this PR

```diff
diff --git a/src/persist/src/lib.rs b/src/persist/src/lib.rs
index 4aba97fac..92e02d0ab 100644
--- a/src/persist/src/lib.rs
+++ b/src/persist/src/lib.rs
@@ -10,11 +10,7 @@
 //! Persistence for Materialize dataflows.
 
 #![warn(missing_docs)]
-#![warn(
-    clippy::cast_possible_truncation,
-    clippy::cast_precision_loss,
-    clippy::cast_sign_loss
-)]
+#![warn(clippy::as_conversions)]
 
 pub mod error;
 pub mod file;
@@ -131,3 +127,8 @@ impl<W, M> Token<W, M> {
         (self.write, self.meta)
     }
 }
+
+/// WIP
+pub fn wipfoo() {
+    let _ = 0usize as u64;
+}

```